aisuite/providers/watsonx_provider.py

```diff
@@ -0,0 +1,76 @@
+import os
+import httpx
+from aisuite.provider import Provider, LLMError
+from aisuite.framework import ChatCompletionResponse
+
+
+class WatsonxProvider(Provider):
+    """
+    WatsonX AI Provider using httpx for direct API calls.
+    """
+
+    cluster_url = "eu-de.ml.cloud.ibm.com"
+    BASE_URL = f'https://{cluster_url}/ml/v1/text/chat?version=2023-10-25'
+
+    def __init__(self, **config):
+        """
+        Initialize the WatsonX provider with the given configuration.
+        The API key is fetched from the config or environment variables.
+        """
+        self.api_key = config.get("api_key", os.getenv("IBM_ACCESS_TOKEN"))
+        self.project_id = config.get("watsonx_project_id", os.getenv("WATSONX_PROJECT_ID"))
+        self.cluster_url = config.get("cluster_url", os.getenv("WATSONX_CLUSTER_URL"))
+
+        if not self.api_key:
+            raise ValueError(
+                "WatsonX API key is missing. Please provide it in the config or set the WATSONX_API_KEY environment variable."
+            )
+        
+        if not self.project_id:
+            raise ValueError(
+                "WatsonX Project ID is missing. Please provide it in the config or set the WATSONX_PROJECT_ID environment variable."
+            )
+        
+        # Optionally set a custom timeout (default to 30s)
+        self.timeout = config.get("timeout", 30)
+
+    def chat_completions_create(self, model, messages, **kwargs):
+        """
+        Makes a request to the WatsonX AI chat completions endpoint using httpx.
+        """
+        headers = {
+            "Authorization": f"Bearer {self.api_key}",
+            "Content-Type": "application/json",
+        }
+
+        data = {
+            "model_id": model,
```

Comment:
model_id : model
this is kinda confusing. am I passing an entire model to an id ?
you might want to rename for clarity and consistency

Reply:
The change should come from the interface, I'm only implementing it unfortunately